addons/stock_account/models/product.py

```diff
@@ -627,6 +627,10 @@ def _svl_replenish_stock_am(self, stock_valuation_layers):
             debit_account_id = product_accounts[product.id]['stock_valuation'].id
             credit_account_id = product_accounts[product.id]['stock_input'].id
             value = out_stock_valuation_layer.value
+            if out_stock_valuation_layer.currency_id.compare_amounts(value, 0) < 0:
+                # Swap accounts makes a negative value in accounting
+                debit_account_id, credit_account_id = credit_account_id, debit_account_id
```

Comment:
If you have a negative quantity, you should probably balance output and not input